src/pages/DownloadScreen.tsx

```diff
@@ -1,45 +1,35 @@
-import { IonButton, IonButtons, IonContent, IonHeader, IonInput, IonItem, IonModal, IonToolbar } from '@ionic/react'
+import { IonButton, IonButtons, IonContent, IonHeader, IonInput, IonItem, IonModal, IonText, IonToolbar } from '@ionic/react'
 import { OfflineMapContainer } from '../components/OfflineMapContainer'
-import { Circle, Marker, Popup, Rectangle, TileLayer, FeatureGroup } from 'react-leaflet'
+import { Circle, Marker, Popup } from 'react-leaflet'
+import React, { useState } from 'react'
 import MarkerHelper from '../helpers/MarkerHelper'
-import React, { useEffect, useState } from 'react'
 import { useLocation } from '../hooks/useLocation'
-import { EditControl } from 'react-leaflet-draw'
-import L, { LatLngBounds } from 'leaflet'
 import { LoadingIndicator } from '../components/LoadingIndicator'
 import { AppScreen } from '../components/AppScreen'
 import { useLandmarks } from '../hooks/useLandmarks'
 import { WarningPopup } from '../components/WarningPopup'
+import { useMap } from '../contexts/MapContext'
+
+import '../components/css/DownloadScreen.css'
 
 const pageName = 'Download landmarks'
 
 export function DownloadScreen() {
   const currentPosition = useLocation()
+  const map = useMap()
 
   const { downloadNewLandmarks, isDownloadSuccess, isErrorDownloadingLandmark, isDownloadingLandmark, sourceName } = useLandmarks('', '')
 
-  const [rectangleBounds, setRectangleBounds] = useState<LatLngBounds>()
   const [areaName, setAreaName] = useState<string>('')
   const [isModalOpen, setIsModalOpen] = useState(false)
 
-  const locationNotEnabled = currentPosition.lat === 0 && currentPosition.lng === 0
-
-  useEffect(() => {
-    if (!rectangleBounds && currentPosition.lng !== 0 && currentPosition.lat !== 0) {
-      setRectangleBounds(
-        new L.LatLngBounds(
-          [currentPosition.lat - 0.005, currentPosition.lng - 0.005],
-          [currentPosition.lat + 0.005, currentPosition.lng + 0.005]
-        )
-      )
-    }
-  }, [currentPosition])
+  const locationEnabled = !(currentPosition.lat === 0 && currentPosition.lng === 0)
 
   function handleDownload() {
-    if (rectangleBounds && areaName) {
+    if (areaName && map && map.current) {
       downloadNewLandmarks({
         areaName: areaName,
-        boundingBox: { topLeft: rectangleBounds.getNorthWest(), bottomRight: rectangleBounds.getSouthEast() },
+        boundingBox: { topLeft: map.current.getBounds().getNorthWest(), bottomRight: map.current.getBounds().getSouthEast() },
```

Comment:
I should create another issue to make clear that this is not functioning properly.

Reply:
Created #68